Doesn't compile with Clang

Primehack doesn't compile with Clang, but does compile with Gcc on my computer (Linux NixOS).
Sorry if that's all I got, I tried to make a quick development environment to investigate, but I'm a beginner and just made a mess.
Anyway, these are the logs. I think that it's related to this: https://stackoverflow.com/questions/63498058/structure-binding-followed-by-a-lambda-capture-creates-a-clang-compilation-error

```
/build/source/Source/Core/Core/PrimeHack/Mods/ElfModLoader.cpp:520:52: error: reference to local binding 'hook_addr' declared in enclosing function 'prime::ElfModLoader::parse_and_load_modfile'
        create_vthook_callgated(hook_sym->address, hook_addr);
                                                   ^
/build/source/Source/Core/Core/PrimeHack/Mods/ElfModLoader.cpp:518:25: note: 'hook_addr' declared here
  for (auto&& [hook_fn, hook_addr] : parsed_vthooks) {
                        ^
/build/source/Source/Core/Core/PrimeHack/Mods/ElfModLoader.cpp:525:28: error: 'hook_addr' in capture list does not name a variable
    resolve_symbols([this, hook_addr] (Symbol* hook_sym) {
                           ^
/build/source/Source/Core/Core/PrimeHack/Mods/ElfModLoader.cpp:526:52: error: reference to local binding 'hook_addr' declared in enclosing function 'prime::ElfModLoader::parse_and_load_modfile'
        create_blhook_callgated(hook_sym->address, hook_addr);
                                                   ^
/build/source/Source/Core/Core/PrimeHack/Mods/ElfModLoader.cpp:524:25: note: 'hook_addr' declared here
  for (auto&& [hook_fn, hook_addr] : parsed_blhooks) {
                        ^
/build/source/Source/Core/Core/PrimeHack/Mods/ElfModLoader.cpp:531:28: error: 'hook_addr' in capture list does not name a variable
    resolve_symbols([this, hook_addr] (Symbol* hook_sym) {
                           ^
/build/source/Source/Core/Core/PrimeHack/Mods/ElfModLoader.cpp:532:56: error: reference to local binding 'hook_addr' declared in enclosing function 'prime::ElfModLoader::parse_and_load_modfile'
        create_trampoline_callgated(hook_sym->address, hook_addr);
                                                       ^
/build/source/Source/Core/Core/PrimeHack/Mods/ElfModLoader.cpp:530:25: note: 'hook_addr' declared here
  for (auto&& [hook_fn, hook_addr] : parsed_trampolines) {
```

--- a/Source/Core/Core/PrimeHack/Mods/ElfModLoader.cpp
+++ b/Source/Core/Core/PrimeHack/Mods/ElfModLoader.cpp
@@ -516,19 +516,19 @@ void ElfModLoader::parse_and_load_modfile(std::string const& path) {
   // TODO: check if too many callgate table entries
   // should never go over really, but good for clarity
   for (auto&& [hook_fn, hook_addr] : parsed_vthooks) {
-    resolve_symbols([this, hook_addr] (Symbol* hook_sym) {
+    resolve_symbols([this, hook_addr = hook_addr] (Symbol* hook_sym) {
         create_vthook_callgated(hook_sym->address, hook_addr);
       }, hook_fn);
   }
 
   for (auto&& [hook_fn, hook_addr] : parsed_blhooks) {
-    resolve_symbols([this, hook_addr] (Symbol* hook_sym) {
+    resolve_symbols([this, hook_addr = hook_addr] (Symbol* hook_sym) {
         create_blhook_callgated(hook_sym->address, hook_addr);
       }, hook_fn);
   }
 
   for (auto&& [hook_fn, hook_addr] : parsed_trampolines) {
-    resolve_symbols([this, hook_addr] (Symbol* hook_sym) {
+    resolve_symbols([this, hook_addr = hook_addr] (Symbol* hook_sym) {
         create_trampoline_callgated(hook_sym->address, hook_addr);
       }, hook_fn);
   }

